### DISCUSSION TITLE
Update repository code

### DISCUSSION THREAD

**Discussion by rquevedo** [Tecnativa/doodba#388] [Q&A] [answered]
Hi folks,
Im trying to update repositories code inside a odoo container. My firts idea was run:
```
docker-compose up --build --force-recreate --no-deps -d odoo
```
But this command dont update repositories and seems to do nothing because all outputs shows "From cache ...".
Which is the rigth or supported way to update source code?
Thanks in advance

**Comment by pedrobaeza** [comment, marked as answer]
`docker-compose build --pull --no-cache`

**Reply by rquevedo** [reply]
Thanks @pedrobaeza this way works like a charm.

**Reply by rquevedo** [reply]
Hi @pedrobaeza In my previous comment i say that command:
`
docker-compose build --pull --no-cache
`
works, and it works one time. But other time even when rebuild entire image code dont get updated. After many test i realize that works when any change has been added to addons.yaml or repos.yaml, if those files dont change code dont  get updated. In my case i dont want add new repos or new addons only update git code inside container.

**Reply by pedrobaeza** [reply]
The `--no-cache` part rebuilds from scratch no matter if any change is done, so if not working, you are not doing something the same I indicated you.

**Comment by joao-p-marques** [comment]
If you are using https://github.com/Tecnativa/doodba-copier-template and you want to update the Odoo or OCA code, you can also run `invoke git-aggregate`, which downloads the latest changes from the Git repos, without rebuilding the images, though.

**Reply by rquevedo** [reply]
@joao-p-marques Im using copier-template and i try your sugestion. Seems like aggregator get executed but inside container the code remains outdated. This way (invoke git-aggregate) should update code inside docker container? Or is for develop environment?

**Reply by pedrobaeza** [reply]
git-aggregate is for development environment.
